### PR TITLE
SCHEMA: Update references to use _ref instead of $ref

### PR DESCRIPTION
**Pull request by effigies**
It looks like using `$ref` will cause issues with the MATLAB YAML reader (or maybe just MATLAB). I think `_ref` is very unlikely to hit collisions while being friendly to various languages.

### CODE REVIEW

**Review by tsalo — APPROVED** [review]
LGTM!

**Comment by Remi-Gau** [comment]
> It looks like using `$ref` will cause issues with the MATLAB YAML reader (or maybe just MATLAB). I think `_ref` is very unlikely to hit collisions while being friendly to various languages.

I hate to be that guy so let me bring in some self-deprecation into the mix...

Depends if you consider MATLAB a language...

> MATLAB YAML reader

If only there was such a thing... 

YAML pretty much has to be converted to JSON for MATLAB

AFAIK

> `_ref` is very unlikely to hit collisions

Well loading a JSON that with such key in MATLAB will very likely lead to problems

Added a section in the starter kit because it may matter for some aspects of BIDS events file: 
https://bids-standard.github.io/bids-starter-kit/folders_and_files/metadata.html#interoperability-issues

the TL;DR for here is:

- JSON (YAML in this case) keys must start with a letter

**Review by Remi-Gau — CHANGES_REQUESTED** [review]
> It looks like using `$ref` will cause issues with the MATLAB YAML reader (or maybe just MATLAB). I think `_ref` is very unlikely to hit collisions while being friendly to various languages.

I hate to be that guy so let me bring in some self-deprecation into the mix...

Depends if you consider MATLAB a language...

> MATLAB YAML reader

If only there was such a thing... 

YAML pretty much has to be converted to JSON for MATLAB

AFAIK

> `_ref` is very unlikely to hit collisions

Well loading a JSON that with such key in MATLAB will very likely lead to problems

Added a section in the starter kit because it may matter for some aspects of BIDS events file: 
https://bids-standard.github.io/bids-starter-kit/folders_and_files/metadata.html#interoperability-issues

the TL;DR for here is:

- JSON (YAML in this case) keys must start with a letter

**Comment by effigies** [comment]
Maybe instead of this, we should write a `schema2json` tool that will take our YAML schema with references in it and produce a fully rendered schema with no references. If a conversion is already necessary, why implement dereferencing in another language?

**Comment by Remi-Gau** [comment]
> Maybe instead of this, we should write a `schema2json` tool that will take our YAML schema with references in it and produce a fully rendered schema with no references. If a conversion is already necessary, why implement dereferencing in another language?

uuuuhhh I like that :+1: 

Also would make the fully rendered schema more explicit (albeit more verbose, no?) which could be good for new users.

**Comment by effigies** [comment]
Okay, let's discuss this tomorrow.
